pipeline/deploy.sh

```diff
@@ -15,31 +15,102 @@ then
 fi
 
 # Borrow a testbed from CSI Testbed Pool Svc.
-if ! testbed=$(curl -X 'GET' "${CNS_TESTBEDPOOL_SVC_URL}");
+if ! testbed=$(curl -X 'PUT' "${CNS_TESTBEDPOOL_SVC_URL}" -H 'accept: application/json' -u "${CNS_MANAGER_USERNAME}:${CNS_MANAGER_PASSWORD}");
 then
 	echo "Unable to borrow a testbed!"
 	exit 1
 fi
+
 echo "TestbedInfo: $testbed"
 
+# Extract testbed ID from borrow testbed API Response
+if ! id=$(echo "$testbed" | jq '.id'|tr -d '"');
+then
+	echo "Error getting the id!"
+	exit 1
+fi
+
+# Extract VC IP from borrow testbed API Response
 if ! vcIp=$(echo "$testbed" | jq '.vcIp'|tr -d '"');
 then
 	echo "Error getting the vcIp!"
 	exit 1
 fi
 
-echo "VC_IP=$vcIp" >> build.env
+# Extract External Gateway VM IP from borrow testbed API Response
+if ! externalVMGatewayIp=$(echo "$testbed" | jq '.externalVMGatewayIp'|tr -d '"');
+then
+	echo "Error getting the externalVMGatewayIp!"
+	exit 1
+fi
 
-if ! svAdminCreds=$(echo "$testbed" | jq '.svAdminCreds'|tr -d '"'|base64 -d);
+
+# Extract VC root password from borrow testbed API Response
+if ! vcRootPassword=$(echo "$testbed" | jq '.vcRootPassword'|tr -d '"');
 then
-	echo "Error getting the svAdminCreds!"
+	echo "Error getting the vcRootPassword!"
 	exit 1
 fi
-echo "svAdminCreds = $svAdminCreds"
+
+# Extract VC username from borrow testbed API Response
+if ! vimUsername=$(echo "$testbed" | jq '.vimUsername'|tr -d '"');
+then
+	echo "Error getting the vimUsername!"
+	exit 1
+fi
+
+# Extract VC Admin password from borrow testbed API Response
+if ! vimPassword=$(echo "$testbed" | jq '.vimPassword'|tr -d '"');
+then
+	echo "Error getting the vimPassword!"
+	exit 1
+fi
+
+# Print all the values into Console
+echo "VSPHERE_CSI_CONTROLLER_IMAGE = $VSPHERE_CSI_CONTROLLER_IMAGE"
+echo "VSPHERE_SYNCER_IMAGE = $VSPHERE_SYNCER_IMAGE"
+echo "Testbed id=$id"
+echo "VC_IP=$vcIp"
+echo "vcRootPassword=$vcRootPassword"
+echo "vimPassword=$vimPassword"
+echo "vimUsername=$vimUsername"
+echo "externalVMGatewayIp=$externalVMGatewayIp"
+
+
+# Store all the values into Artifacts
+echo "id=$id" >> ./env.json
```

Comment:
Why we need 2 artifacts - `env.json` and `build.env`? Can we consolidate them into one?

Reply:
Added you to a slack discussion - let me know if you have any more questions